### PR TITLE
Added the aircraft carrier and its hornets

### PR DESCRIPTION
**Pull request by Mailaender**
Adds the code from @forcecore.

Closes https://github.com/OpenRA/ra2/issues/151.
Closes https://github.com/OpenRA/OpenRA/issues/4157.

### CODE REVIEW

**Comment by ABrandau** [comment]
Do note that I have been using this code for more than a year, and it has a bug where aircraft fail to land on the carrier, becoming stuck in place, and unable to get linked to the carrier again.

**Comment by Mailaender** [comment]
This could be the same issue as https://github.com/OpenRA/OpenRA/issues/14365.

File: OpenRA.Mods.RA2/Traits/BaseSpawnMaster.cs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+	{
+		[Desc("Spawn these units. Define this like paradrop support power.")]
+		public readonly string[] Actors;
+
+		[Desc("Slave actors to contain upon creation. Set to -1 to start with full slaves.")]
+		public readonly int InitialActorCount = -1;
+
+		[Desc("The armament which will trigger the slaves to attack the target. (== \"Name:\" tag of Armament, not @tag!)")]
+		[WeaponReference]
+		public readonly string SpawnerArmamentName = "primary";
```
Comment (MustaphaTR) [inline OpenRA.Mods.RA2/Traits/BaseSpawnMaster.cs]: This isn't a WeaponReference and causing the Travis to fail.

**Comment by MustaphaTR** [comment]
#508

**Pull request closed without merging**